packages/cli-repl/test/e2e-direct.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { MongodSetup, startTestCluster } from '../../../testing/integration-testing-hooks';
+import { eventually } from './helpers';
+import { TestShell } from './test-shell';
+
+describe('e2e direct connection', () => {
+  afterEach(async() => await TestShell.killall());
+
+  context('to a replica set', async() => {
+    const replSetId = 'replset';
+    const [rs0, rs1, rs2] = startTestCluster(
+      ['--single', '--replSet', replSetId],
+      ['--single', '--replSet', replSetId],
+      ['--single', '--replSet', replSetId]
+    );
+
+    [ [rs0, 'rs0'], [rs1, 'rs1'], [rs2, 'rs2'] ].forEach((elem) => {
+      const server = elem[0] as MongodSetup;
+      const node = elem[1] as string;
```

Comment:
With something like

```suggestion
    for (const { server, node } of [
      { server: rs0, node: 'rs0' },
      { server: rs1, node: 'rs1' },
      { server: rs2, node: 'rs2' }
    ]) {
```

you should be able to avoid the type casting here :)

Reply:
(unfortunately I got a TypeScript error here so explicitly added it - will check again :/ )

Reply:
Yup works when making it an object 👍